Makefile

```diff
@@ -0,0 +1,86 @@
+.DEFAULT_GOAL := help
+
+.PHONY: help
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+# -- variables --------------------------------------------------------------------------------------
+
+WARNINGS=RUSTDOCFLAGS="-D warnings"
+DEBUG_OVERFLOW_INFO=RUSTFLAGS="-C debug-assertions -C overflow-checks -C debuginfo=2"
+
+# -- linting --------------------------------------------------------------------------------------
+
+.PHONY: clippy
+clippy: ## Runs Clippy with configs
+	$(WARNINGS) cargo +nightly clippy --workspace --all-targets --all-features
+
+
+.PHONY: fix
+fix: ## Runs Fix with configs
+	cargo +nightly fix --allow-staged --allow-dirty --all-targets --all-features
+
+
+.PHONY: format
+format: ## Runs Format using nightly toolchain
+	cargo +nightly fmt --all
+
+
+.PHONY: format-check
+format-check: ## Runs Format using nightly toolchain but only in check mode
+	cargo +nightly fmt --all --check
+
+
+.PHONY: lint
+lint: format fix clippy ## Runs all linting tasks at once (Clippy, fixing, formatting)
+
+# --- docs ----------------------------------------------------------------------------------------
+
+.PHONY: doc
+doc: ## Generates & checks documentation
+	$(WARNINGS) cargo doc --all-features --keep-going --release
+
+# --- testing -------------------------------------------------------------------------------------
+
+.PHONY: test-default
+test-default: ## Run default tests
+	$(DEBUG_OVERFLOW_INFO) cargo nextest run --profile default --release --all-features
+
+
+.PHONY: test-no-default
+test-no-default: ## Run tests with `no-default-features`
+	$(DEBUG_OVERFLOW_INFO) cargo nextest run --profile default --release --no-default-features
```

Comment:
I'm wondering if this should be `test-no-std` to be consistent with `build-no-std`. But also, maybe it is fine to keep as is.

Reply:
Updated it to `test-no-std` considering that the only thing that is enabled by the `default-features` is `std`